src/mongocxx/options/server_api.hpp

```diff
@@ -0,0 +1,142 @@
+// Copyright 2021 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <string>
+
+#include <bsoncxx/stdx/optional.hpp>
+#include <bsoncxx/stdx/string_view.hpp>
+#include <mongocxx/stdx.hpp>
+
+#include <mongocxx/config/prelude.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+class client;
+class pool;
+
+namespace options {
+
+///
+/// Class representing options for server API.
+///
+class MONGOCXX_API server_api {
+   public:
+    ///
+    /// Enum representing the possible values for server API version.
+    ///
+    enum class version { k_version_1 };
+
+    ///
+    /// Converts a version enum value to its string value.
+    ///
+    /// @param version
+    ///   The enum value to convert to a string.
+    ///
+    /// @return
+    ///   The string value of the given enum value.
```

Comment:
Minor: Note that these methods can throw with `@throws mongocxx::logic_error on an invalid argument`

Reply:
Added here and on `version_from_string`.